apiserver/facades/client/charms/charmhubrepo.go

```diff
@@ -328,23 +328,24 @@ func matchesType(originType string, expected transport.Type) bool {
 }
 
 func composeSuggestions(releases []transport.Release, origin params.CharmOrigin) []string {
-	var suggestions []string
+	channelSeries := make(map[string][]string, 0)
 	for _, release := range releases {
 		platform := release.Platform
-		arch, os, series := platform.Architecture, platform.OS, platform.Series
+		arch, series := platform.Architecture, platform.Series
 		if arch == "all" {
 			arch = origin.Architecture
 		}
 		if arch != origin.Architecture {
 			continue
 		}
-		if os == "all" {
-			os = origin.OS
-		}
 		if series == "all" {
 			series = origin.Series
 		}
-		suggestions = append(suggestions, fmt.Sprintf("%s with %s/%s/%s", release.Channel, arch, os, series))
+		channelSeries[release.Channel] = append(channelSeries[release.Channel], series)
+	}
+	var suggestions []string
+	for channel, values := range channelSeries {
+		suggestions = append(suggestions, fmt.Sprintf("%s with %s", channel, strings.Join(values, ", ")))
```

Comment:
We should sort the order of the suggestions. I'd say put latest first then sort everything after it in a natural ordering.

The tests are failing because this can be any order because of using a map to index channels.

Reply:
updated.